### PR TITLE
Make dark mode restartless

### PR DESCRIPTION
**Pull request by ozone10**
Make dark mode restartless.

Should work for everything, even treeview scrollbars.
fix https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10121

@donho 
I have just noticed that Notepad++ still does not allow darkmode in Windows 11.
Should I do commit to allow it or wait for stable  Windows 11 build.

### CODE REVIEW

File: PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp [hunk 1]
```diff
@@ -451,34 +453,34 @@ bool ProjectPanel::saveWorkSpace()
...
+	tvItem.mask = TVIF_TEXT;
+	tvItem.pszText = textBuffer;
+	tvItem.cchTextMax = MAX_PATH;
+
+	//for each project, write <Project>
+	HTREEITEM tvRoot = _treeView.getRoot();
+	if (!tvRoot)
+		return false;
+
+		for (HTREEITEM tvProj = _treeView.getChildFrom(tvRoot);
```
Comment (donho) [inline PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp:472]: Not aligned correctly.

**Comment by donho** [comment]
@ozone10 
Awesome!

> I have just noticed that Notepad++ still does not allow darkmode in Windows 11.
Should I do commit to allow it or wait for stable Windows 11 build.

Does it need a lot of code for the activation in Windows 11? Is there any impact for dark mode in Windows 10 if you add activation code?

File: PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp (hunk 1 above)
Comment (ozone10) [reply PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp:472]: Strange I did automatic tabify in VS2019, will fix it.

File: PowerEditor/src/DarkMode/DarkMode.cpp [hunk 2]
```diff
@@ -241,17 +249,17 @@ constexpr bool CheckBuildNumber(DWORD buildNumber)
 		buildNumber == 19043);  // 21H1
 }
```
Comment (ozone10) [inline PowerEditor/src/DarkMode/DarkMode.cpp:250]: code to change to make darkmode on Windows 11

**Comment by ozone10** [comment]
> Does it need a lot of code for the activation in Windows 11? Is there any impact for dark mode in Windows 10 if you add activation code for Windows 11?

nope I only need to allow it in
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/eac8c4094f93bbd50e9eb29084ca001eee820b0b/PowerEditor/src/DarkMode/DarkMode.cpp#L234

change this `return (buildNumber == 17763 || // 1809` to `return (buildNumber >= 17763);` to allow every builds
or `buildNumber == 19043);  // 21H1` to `return (buildNumber >= 19043); to allow only latest insider
or add build number 22000, to allow only current insider.

**Comment by donho** [comment]
> nope I only need to allow it in

OK, please do it in another PR with the PR title "Activate dark mode in Windows 11".

File: PowerEditor/src/DarkMode/DarkMode.h [hunk 3]
```diff
@@ -13,5 +13,5 @@ bool IsColorSchemeChangeMessage(LPARAM lParam);
 bool IsColorSchemeChangeMessage(UINT message, LPARAM lParam);
 void AllowDarkModeForApp(bool allow);
 void EnableDarkScrollBarForWindowAndChildren(HWND hwnd);
-void InitDarkMode(bool fixDarkScrollbar, bool dark);
-
+void InitDarkMode();
```
Comment (sheenkhan) [inline PowerEditor/src/DarkMode/DarkMode.h:16]: -

**Pull request closed without merging**